pkgs/development/zig-packages/build-zig-package/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib, stdenv, zig, mktemp }:
+
+{ strictDeps ? true
+, depsBuildBuild ? [ ]
+, nativeBuildInputs ? [ ]
+, configurePhase ? null
+, buildPhase ? null
+, checkPhase ? null
+, installPhase ? null
+, enableParallelBuilding ? true
+, meta ? { }
+, ...
+}@attrs:
+
+stdenv.mkDerivation (attrs // {
+  inherit strictDeps enableParallelBuilding;
+  depsBuildBuild = [ ] ++ depsBuildBuild;
+  nativeBuildInputs = [ zig ] ++ nativeBuildInputs;
+
+  configurePhase =
+    if (configurePhase == null) then ''
+      runHook preConfigure
+      runHook postConfigure
+    '' else
+      configurePhase;
+
+  postConfigure = ''
+    export XDG_CACHE_HOME=$(mktemp -d)
+  '';
+
+  buildPhase =
+    if (buildPhase == null) then ''
+      runHook preBuild
+      zig build --verbose
```

Comment:
Why are we double `zig build`-ing?

Reply:
It doesnt build it the second time. The install command is behind the build command. All it does is move the result to $out.